cmd/bootnode/main.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 The AthanorLabs/atomic-swap Authors
+// SPDX-License-Identifier: LGPL-3.0-only
+
+// Package main provides the entrypoint of the bootnode executable,
+// a node that is only used to bootstrap the p2p network and does not run
+// any swap services.
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/athanorlabs/atomic-swap/cliutil"
+	"github.com/athanorlabs/atomic-swap/common"
+	"github.com/athanorlabs/atomic-swap/net"
+	"github.com/athanorlabs/atomic-swap/rpc"
+
+	"github.com/hashicorp/go-multierror"
+	logging "github.com/ipfs/go-log"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	defaultLibp2pPort = 9900
+	defaultRPCPort    = common.DefaultSwapdPort
+
+	flagDataDir    = "data-dir"
+	flagLibp2pKey  = "libp2p-key"
+	flagLibp2pPort = "libp2p-port"
+	flagBootnodes  = "bootnodes"
+	flagRPCPort    = "rpc-port"
+	flagEnv        = "env"
+)
+
+var log = logging.Logger("cmd")
+
+func cliApp() *cli.App {
+	return &cli.App{
+		Name:                 "bootnode",
+		Usage:                "A bootnode for the atomic swap p2p network.",
+		Version:              cliutil.GetVersion(),
+		Action:               runBootnode,
+		EnableBashCompletion: true,
+		Suggest:              true,
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  flagDataDir,
+				Usage: "Path to store swap artifacts",
+				Value: "{HOME}/.atomicswap/{ENV}", // For --help only, actual default replaces variables
```

Comment:
I'm looking at `docs/default_file_locations.md` and I forgot to update the relayer section when I removed the separate relayer binary. Let's repurpose that section for the bootnode.

Should we put the dedicated bootnode's files in `{home}/.atomicbootnode/{ENV}` or something similar? Our failure modes when 2 programs try to access the same DB at the same time could be cleaner. The fewer defaults that testers have to override to run a swapd and bootnode instance on the same host the better.

I'm wondering if we should use a different default port as well? Or have a default bootnode port, but use an OS assigned random port for clients by default instead of a fixed port.

Reply:
right, yeah i was thinking of putting it in `.atomicswap/bootnode`, can update that

a different libp2p port would be fine, but I don't think it matters hugely since a user can just change it on the command line, what do you think?